File: 이진영/Stack/주식가격.js
function solution(prices) {
    let l = prices.length;
    let answer = Array(l).fill(0);
    let s = [];
    
    for (let i = 0; i < l; i++) {
        while (s.length > 0 && prices[i] < s[s.length - 1][1]) {
            let pop = s.pop();
            answer[pop[0]] = i - pop[0];
        }
        s.push([i, prices[i]]);
    }
    
    while (s.length > 0) {
        let pop = s.pop();
        answer[pop[0]] = l - 1 - pop[0];
    }
    
    return answer;
}

File: 정민선/프로그래머스/주식가격.py
def solution(prices):
    answer = []
    n = len(prices)
    for i in range(n):
        count = 0
        for j in range(i+1, n):
            count += 1
            if prices[j] < prices[i]:
                break
        answer.append(count)
    
    return answer